### PR TITLE
[cli] fix usage of strncpy in coap

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 👍

File: src/cli/cli_coap_secure.cpp [hunk 1]
```diff
@@ -249,8 +250,9 @@ otError CoapSecure::ProcessRequest(Arg aArgs[], otCoapCode aCoapCode)
 #endif
 
     VerifyOrExit(!aArgs[0].IsEmpty(), error = OT_ERROR_INVALID_ARGS);
-
+    VerifyOrExit(aArgs[0].GetLength() < kMaxUriLength, error = OT_ERROR_INVALID_ARGS);
     strncpy(coapUri, aArgs[0].GetCString(), sizeof(coapUri) - 1);
+    coapUri[sizeof(coapUri) - 1] = '\0';
```
Comment (simonlingoogle) [inline src/cli/cli_coap_secure.cpp:255]:
```suggestion
    VerifyOrExit(aArgs[0].GetLength() < sizeof(coapUri), error = OT_ERROR_INVALID_ARGS);
    strncpy(coapUri, aArgs[0].GetCString(), sizeof(coapUri));
```

Comment (jwhui) [reply src/cli/cli_coap_secure.cpp:255]: How about just using `strcpy`?

Comment (simonlingoogle) [reply src/cli/cli_coap_secure.cpp:255]: SGTM

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Pull request merged**